### PR TITLE
Fixed Single Store Mode - Incorrect scope values using at Storefront for Catalog Section

### LINKED ISSUE #25933: Single Store Mode- Incorrect scope values using at Storefront for Catalog Section
### Preconditions (*)

1. Magento 2.3.3 CE & 2.4-develop with sample data
2. Magento Instance should have one **Website**, **Store** and **Store View** only.

### Steps to reproduce (*)

1. Under Default Config, Navigate to Stores > Settings > Configuration > Catalog > Catalog
2. Change the values of field Products per Page on Grid Allowed Values from **9,15,30** to **5,10,20,40,80** (Uncheck **Use System value** checkbox if ticked - optional)
3. Change the values of field Products per Page on Grid Default Value from **9** to **10**  (Uncheck **Use System value** checkbox if ticked- optional)
4. Click on **Save Config**
5. Clear the cache if prompted to refresh the cache
6. Switch the scope from **Default Config** to **Default Store View** Scope from the drop-down showing on upper-left corner
7. Navigate to Stores > Settings > Configuration > Catalog > Catalog
8. From Default Store View Scope, Uncheck **Use Website** checkbox for following fields
- Products per Page on Grid Allowed Values
- Products per Page on Grid Default Value
9.  Change the values of field Products per Page on Grid Allowed Values to 30,60,90
10. Change the values of field Products per Page on Grid Default Value	to 30
11. Click on **Save Config**
11. Switch the scope to **Default Config** from **Default Store View Scope**
12. From Default Config Scope, Set Enable Single-Store Mode to **Yes**
13. Click **Save Config**
14. Clear all the cache
15. Now, Navigate to Stores > Settings > Configuration > Catalog > Catalog
16. Tries to update the values for following fields.
- Products per Page on Grid Allowed Values **20,40,60**
- Products per Page on Grid Default Value **20**
17. **Save Config** and Clear all the cache
18. Open any category page to see the **Show per page** values.

### Expected result (*)

1. On category page, it should show 20,40,60 in the **Show per page** drop-down.

### Actual result (*)

1. On category page, it shows values as 30,60,90  in the **Show per page** drop-down.

### Additional Info
1. The issue may be reproduced for all other fields which are available on `showInStore=1`

### PR DESCRIPTION
**Pull request by muthu-ziffity**
### Description (*)

In category page, In single store mode Show per page section doesn't get the value from admin properly. Now it got fixed in this Pull Request

### Fixed Issues (if relevant)

1. Fixes magento/magento2#25933

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by rogyar — CHANGES_REQUESTED** [review]
Hi @muthu-ziffity. Thank you for your contribution. As far as I see, the original issue says that this issue is reproducible for all issues that have `showInStore` flag. I'm not sure that fixing one particular occasion solves the problem. 
Also, the single store mode should not affect the area from where we get the configuration. The single store mode only "hides" the possibility to switch scopes in the admin panel.

**Review by muthu-ziffity — COMMENTED** [review]
I have provided the fix for getting config value globally while single store mode was enabled and added test case for that.

Find below video have the output of fix

output

**Comment by muthu-ziffity** [comment]
> Hi @muthu-ziffity. Thank you for your contribution. As far as I see, the original issue says that this issue is reproducible for all issues that have `showInStore` flag. I'm not sure that fixing one particular occasion solves the problem.
> Also, the single store mode should not affect the area from where we get the configuration. The single store mode only "hides" the possibility to switch scopes in the admin panel.

Provided a fix for this issue globally. Please review again

**Review by muthu-ziffity — COMMENTED** [review]
@akaplya @rogyar Please review this

**Comment by rogyar** [comment]
Hi @muthu-ziffity. I'm putting it on hold for now. This solution should be discussed internally for the final decision. Thank you for your patience.

**Comment by muthu-ziffity** [comment]
> Hi @muthu-ziffity. I'm putting it on hold for now. This solution should be discussed internally for the final decision. Thank you for your patience.

Ok @rogyar. Awaiting for your final decision of this fix

File: lib/internal/Magento/Framework/App/Config.php
Comment (Stepa4man) [inline lib/internal/Magento/Framework/App/Config.php:99]: Are you sure Framework should be aware of stores qty and such edge cases?

**Review by Stepa4man — COMMENTED** [review]
Comment for group code review

**Comment by engcom-Charlie** [comment]
Hi @muthu-ziffity, I'm closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for your collaboration.

**Pull request closed without merging**
